cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HBaseVersion.java

```diff
@@ -91,6 +94,13 @@ public String getMajorVersion() {
         }
       } else if (versionString.startsWith(HBASE_11_VERSION)) {
         currentVersion = Version.HBASE_11;
+      } else if (versionString.startsWith(HBASE_12_VERSION)) {
+        VersionNumber ver = VersionNumber.create(versionString);
+        if (ver.getClassifier() != null && ver.getClassifier().startsWith(CDH57_CLASSIFIER)) {
+          currentVersion = Version.HBASE_12_CDH57;
+        } else {
+          currentVersion = Version.UNKNOWN;
```

Comment:
Yes. Thats right. The exception is thrown from `HBaseVersionSpecificFactory` - https://github.com/caskdata/cdap/blob/release/3.4/cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HBaseVersionSpecificFactory.java#L57